modules/standard/CTypes.chpl

```diff
@@ -503,8 +609,27 @@ module CTypes {
 
     return c_pointer_return(arr[arr.domain.low]);
   }
+  /*
+   Like c_ptrTo for arrays, but returns a :type:`c_ptrConst` which disallows
+   direct modification of the pointee.
+   */
+  inline proc c_ptrToConst(const arr: []): c_ptrConst(arr.eltType) {
+    if (!arr.isRectangular() || !arr.domain.dist._value.dsiIsLayout()) then
+      compilerError("Only single-locale rectangular arrays support c_ptrToConst() at present");
 
-  /* Returns a :type:`c_ptr` to any Chapel object.
+    if (arr._value.locale != here) then
+      halt("c_ptrToConst() can only be applied to an array from the locale on which it lives (array is on locale " + arr._value.locale.id:string + ", call was made on locale " + here.id:string + ")");
```

Comment:
I turn into a grumpy old man when I see lines exceed 80 columns. I see that you're inheriting this from above. I would be happier if you fix this one and that, too while here. But I can't ask you to do that :)

Reply:
Will do!